### PR TITLE
Option to run LabelModel on GPU

### LINKED ISSUE #1430: Error when using LabelModel with GPU
## Issue description

I tried running LabelModel with `device=torch.device('cuda')` but I get

> RuntimeError: Expected object of backend CUDA but got backend CPU for argument #2 'mat2'

Detailed error message:

> I0827 13:07:07.767502 140001064118080 label_model.py:749] Computing O...
> I0827 13:07:07.818106 140001064118080 label_model.py:755] Estimating \mu...
> I0827 13:07:07.818606 140001064118080 label_model.py:762] Using GPU...
> ---------------------------------------------------------------------------
> RuntimeError                              Traceback (most recent call last)
> <ipython-input-18-b492cdcee02c> in <module>
> ----> 1 label_model.fit(L_train, n_epochs=5000, seed=123, log_freq=20, lr=0.01)
> 
> /etc/anaconda3/envs/ml/lib/python3.7/site-packages/snorkel/labeling/model/label_model.py in fit(self, L_train, Y_dev, class_balance, **kwargs)
>     781 
>     782             # Forward pass to calculate the average loss per example
> --> 783             loss = self._loss_mu(l2=self.train_config.l2)
>     784             if torch.isnan(loss):
>     785                 msg = "Loss is NaN. Consider reducing learning rate."
> 
> /etc/anaconda3/envs/ml/lib/python3.7/site-packages/snorkel/labeling/model/label_model.py in _loss_mu(self, l2)
>     522             Overall mu loss between learned mu and initial mu
>     523         """
> --> 524         loss_1 = torch.norm((self.O - self.mu @ self.P @ self.mu.t())[self.mask]) ** 2
>     525         loss_2 = torch.norm(torch.sum(self.mu @ self.P, 1) - torch.diag(self.O)) ** 2
>     526         return loss_1 + loss_2 + self._loss_l2(l2=l2)
> 
> RuntimeError: Expected object of backend CUDA but got backend CPU for argument #2 'mat2'
> 

## Code example/repro steps

I've added `device=torch.device('cuda')` to line that defines LabelModel  in recsys_tutorial from [this example](https://github.com/snorkel-team/snorkel-tutorials/tree/master/recsys)

## System info

Snorkel installed from [requirements.txt](https://github.com/snorkel-team/snorkel-tutorials/blob/master/recsys/requirements.txt)
* OS: Linux Kubuntu 18.04
* Python version: 3.7.3
* Snorkel version: 0.9.0
* Versions of any other relevant libraries: torch 1.1.0

### PR DESCRIPTION
**Pull request by paroma**
## Description of proposed changes
Fixed some parameter definitions to allow LabelModel to run on GPU.

## Related issue(s)
Fixes #1430 

## Test plan
- Edited existing tests to work with change in parameter definition
- No new tests added. Tested locally by adding `device='cuda'` for LabelModel tests.

## Checklist

Need help on these? Just ask!

* [x] I have read the **CONTRIBUTING** document.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.
* [x] I have run `tox -e complex` and/or `tox -e spark` if appropriate.
* [x] All new and existing tests passed.

### CODE REVIEW

File: snorkel/labeling/model/label_model.py [hunk 1]
```diff
@@ -334,7 +336,7 @@ def get_conditional_probs(self) -> np.ndarray:
         np.ndarray
             An [m, k + 1, k] np.ndarray conditional probabilities table.
         """
-        return self._get_conditional_probs(self.mu.detach().numpy())
+        return self._get_conditional_probs(self.mu.clone().cpu().detach().numpy())
```
Comment (bhancock8) [inline snorkel/labeling/model/label_model.py]: Why is clone() necessary here but not the other places where you're adding `.cpu().detach()`?

File: snorkel/labeling/model/label_model.py [hunk 2]
```diff
@@ -750,7 +752,7 @@ def _count_accurate_lfs(self, mu: np.ndarray) -> int:
         int
             Number of LFs better than random
         """
-        P = self.P.numpy()
+        P = self.P.clone().cpu().detach().numpy()
```
Comment (bhancock8) [inline snorkel/labeling/model/label_model.py]: Same Q.

File: test/labeling/model/test_label_model.py [hunk 3]
```diff
@@ -95,7 +95,9 @@ def test_generate_O(self):
                 [1 / 4, 0, 0, 1 / 4, 0, 1 / 4],
             ]
         )
-        np.testing.assert_array_almost_equal(label_model.O.numpy(), true_O)
+        np.testing.assert_array_almost_equal(
+            label_model.O.cpu().detach().numpy(), true_O
```
Comment (bhancock8) [inline test/labeling/model/test_label_model.py]: Feels very strange to see `label_model.O', but I know that's not part of this PR's scope.

File: snorkel/labeling/model/label_model.py (hunk 1 above)
Comment (paroma) [reply snorkel/labeling/model/label_model.py]: good catch, probably added during debugging. removed

File: snorkel/labeling/model/label_model.py (hunk 2 above)
Comment (paroma) [reply snorkel/labeling/model/label_model.py]: removed

**Pull request merged**